hack/force-cert-retation.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+set -ex
+
+function usage {
+cat <<- EOF
+Usage example:
+  HANDLER_NAMESPACE=kubemacpool-system HANDLER_POD_LABEL=kubemacpool-leader=true WEBHOOK_TYPE=mutatingWebhookConfiguration WEBHOOK_CONFIG_NAME=kubemacpool-mutator WEBHOOK_SECRET_NAME=kubemacpool-service ./force-cert-rotation.sh
+
+Mandatory Arguments:
+  HANDLER_NAMESPACE     namespace of the pod operating the kube-admission's library.
+  HANDLER_POD_LABEL     specific label corresponding to the pod operating the kube-admission's library.
+  WEBHOOK_TYPE          webhook config type: [mutatingWebhookConfiguration|validationWebhookConfiguration].
+  WEBHOOK_CONFIG_NAME   webhook config name.
+  WEBHOOK_SECRET_NAME   the name of the secret created by kube-admission.
+EOF
+}
+
+function input_env_var_check {
+  # check mandatory env variables
+  if [[ -z "${HANDLER_NAMESPACE}" || -z "${HANDLER_POD_LABEL}" || -z "${WEBHOOK_TYPE}" || -z "${WEBHOOK_CONFIG_NAME}" || -z "${WEBHOOK_SECRET_NAME}" ]]; then
+    echo "Error: Not all env parameters are set."
+    usage
+    exit
+  fi
+
+  # check KUBECTL validity
+  if [[ -z "$KUBECTL" || -z "$($KUBECTL version)" ]]; then
+    echo "Error: KUBECTL is not defined"
+    exit
+  fi
+}
+
+# input parameter parsing
+function input_param_parse {
+  while [[ ! -z "$1" ]]; do
+    case "$1" in
+      --help)
+          usage
+          exit
+          ;;
+      *)
+          break
+          ;;
+      esac
+      shift
+  done
+}
+
+# retrive tls.crt from secret data
+function get_secret_data {
+  echo "$(${KUBECTL} get secret -n ${HANDLER_NAMESPACE} ${WEBHOOK_SECRET_NAME} -oyaml | shyaml get-value data)"
```

Comment:
I used shyaml here. not sure if it's the right way to go. what do you think?

Reply:
Unless shyaml is a part of a base Fedora/CentOS distribution, it is a bad idea to use it in a generic script. If it is really that hard to use only `kubectl` commands to handle this, I won't fight against `sed`. However, I think you can iterate all webhook configs with `-o jsonpath='{ ... }'`

Reply:
ack. shyaml is indeed not a good option as it requires seperate pip install.